docs/proposals/scheduling/20220701-schedule-gang.md

```diff
@@ -0,0 +1,156 @@
+---
+title: gang scheduling
+authors:
+- "@buptcozy"
+reviewers:
+- "@hormes"
+- "@allwmh"
+- "@honpey"
+- "@jasonliu747"
+- "@saintube"
+- "@stormgbs"
+- "@zwzhang0107"
+- "@eahydra"
+creation-date: 2022-07-01
+last-updated: 2022-07-01
+status: provisional
+
+---
+
+# gang schedule
+
+## Table of Contents
+
+<!--ts-->
+
+* [gang schedule](#gang-schedule)
+  * [Table of Contents](#table-of-contents)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+  * [Goals](#goals)
+  * [User Stories](#user-stories)
+    * [Story AI](#story-AI)
+  * [Implementation Details](#implementation-details)
+    * [pod annotation to describe gang](#pod-annotation-to-describe-gang)
+    * [gang crd](#gang crd)
+    * [system-design](#system-design)
+
+<!--te-->
+
+## Summary
+gang scheduling
+
+## Motivation
+in AI scenario, we may want worker launch at same time.
+
+### Goals
+1. propose some pod's annotations to announce gang properties
+
+2. design a gang-crd to display gang-status
+
+3. scheduler should provide gang data-structure and schedule-plugin 
+
+### User Stories
+
+#### Story AI
+lots of AI jobs such as tf\pytorch need gang-scheduling
+     
+## Implementation Details
+### pod annotation to describe gang
+we choose to create gang-related by pod's annotations instead of gang crd. This is because we don't want high level
```

Comment:
Could you elaborate more about why we choose annotations but not a CRD?
For instance, what is the case we need a high-level operator to maintain CRD lifecycles, what are the functional differences between the upstream implementation...
We may need more details about the comparisons with the upstream gang scheduling ([co-scheduling](https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/pkg/coscheduling)).

Reply:
updated